frontend/src/app/core/services/auth.service.ts

```diff
@@ -0,0 +1,120 @@
+import { Injectable, NgZone } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
+import { AngularFirestore, AngularFirestoreDocument } from '@angular/fire/compat/firestore';
+import { Router } from '@angular/router';
+import { HttpService } from '@core/services/http.service';
+import { IUser } from '@shared/models/IUser';
+import * as auth from 'firebase/auth';
+import firebase from 'firebase/compat';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class AuthService {
+    private userData: firebase.User;
+
+    private url = '';
+
+    constructor(
+        public afs: AngularFirestore,
+        public afAuth: AngularFireAuth,
+        public router: Router,
+        public ngZone: NgZone,
+        public httpService: HttpService,
+    ) {
+        this.afAuth.authState.subscribe((user) => {
+            if (user) {
+                this.userData = user;
+                localStorage.setItem('user', JSON.stringify(this.userData));
+                JSON.parse(localStorage.getItem('user')!);
+            } else {
+                localStorage.setItem('user', 'null');
+                JSON.parse(localStorage.getItem('user')!);
```

Comment:
Please get rid of duplication codes and you need to unsubscribe after every subscribe

Reply:
you can use removeItem instead of setItem with null